includes/class-sensei-utils.php

```diff
@@ -2487,6 +2487,17 @@ public static function is_rest_request(): bool {
 		return defined( 'REST_REQUEST' ) && REST_REQUEST;
 	}
 
+	/**
+	 * Check if this is a frontend request.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return bool
+	 */
+	public static function is_frontend_request(): bool {
+		return ! self::is_rest_request() && ! is_admin();
```

Comment:
Curious why the call to `is_rest_request` is needed.

Reply:
In this method, it's to get only the frontend requests, ignoring admin requests and rest requests.

For the purpose of the feature of this PR specifically, Gutenberg also makes some REST requests on the editor, which calls the `the_content` filter, calling the `do_blocks`, and consequently persisting the notices.